functions/src/index.ts

```diff
@@ -46,163 +45,194 @@ const genericErrorHandler = (err: any, req: express.Request, res: express.Respon
-  return BaseResourceObject.Find(db, req.env, req.params.id)
+  BaseResourceObject.Find(db, req.env, req.params.id)
     .then(resource => res.success(resource.apiResult()))
     .catch(error => res.error(404, error))
-})
+});
 
 app.post('/api/v1/resources', (req, res) => {
-  return BaseResourceObject.Create(db, req.env, req.claims, req.body)
-    .then(resource => res.success(resource.apiResult(), 201))
-    .catch(error => res.error(400, error))
-})
+  // Note that this is the only route where auth is not called directly. Resource can be created anonymously.
+  const token = getToken(req);
+  if (token) {
+    try {
+      const claims = getClaimsFromJWT(token);
+      BaseResourceObject.Create(db, req.env, claims, req.body)
+        .then(resource => res.success(resource.apiResult(), 201))
+        .catch(error => res.error(400, error));
+    } catch (error) {
+      res.error(403, error);
+    }
+  } else {
+    // Anonymous usage based on readWriteToken. NOT RECOMMENDED apart from testing. readWriteTokens are meant to
+    // be used mostly/only to support converted Document Store documents.
+    BaseResourceObject.Create(db, req.env, undefined, req.body)
+      .then(resource => res.success(resource.apiResult(), 201))
+      .catch(error => res.error(400, error))
+  }
+});
 
 app.patch('/api/v1/resources/:id', (req, res) => {
-  return BaseResourceObject.Update(db, req.env, req.claims, req.params.id, req.body)
-    .then(resource => res.success(resource.apiResult()))
-    .catch(error => res.error(400, error))
-})
+  try {
+    const claimsOrReadWriteToken = authUsingJWTOrReadWriteToken(req);
+    BaseResourceObject.Update(db, req.env, claimsOrReadWriteToken, req.params.id, req.body)
+      .then(resource => res.success(resource.apiResult()))
+      .catch(error => res.error(400, error))
+  } catch (error) {
+
+  }
+});
 
 app.post('/api/v1/resources/:id/credentials', (req, res) => {
-  return getValidatedConfig()
-    .then((config) => BaseResourceObject.CreateAWSKeys(db, req.env, req.claims, req.params.id, config))
-    .then(awsTemporaryCredentials => res.success(awsTemporaryCredentials))
-    .catch(error => res.error(400, error))
-})
+  try {
+    const claimsOrReadWriteToken = authUsingJWTOrReadWriteToken(req);
+    const config = getValidatedConfig();
+    BaseResourceObject.CreateAWSKeys(db, req.env, claimsOrReadWriteToken, req.params.id, config)
+      .then(awsTemporaryCredentials => res.success(awsTemporaryCredentials))
+      .catch(error => res.error(400, error))
+  } catch (error) {
+    res.error(403, error);
+  }
+});
 
 app.delete('/api/v1/resources/:id', (req, res) => {
-  return BaseResourceObject
-    .delete(db, req.env, req.claims, req.params.id, req.body)
-    .then(deleteDate => res.success({date: deleteDate}))
-    .catch(error  => res.error(400, error))
-})
+  try {
+    const claims = authUsingJWT(req); // note that readWriteToken intentionally does NOT give access to delete method
```

Comment:
Yes I think this is the right approach. I don't see a reason to allow readWriteTokens to allow deletion.